week-01/assessment-1/src/main/java/HotelApp.java

```diff
@@ -0,0 +1,343 @@
+//                default:
+//                    System.out.println("Unknown action");
+//                    mainMenu();
+//            }
+
+//        System.out.println("Guest Menu");
+//        System.out.println("====================");
+//        System.out.println("1. Check in");
+//        System.out.println("2. Check out");
+//        System.out.println("3. View Guests");
+//        System.out.println("4. Exit");
+//        System.out.println("Choose an option [1-4]: ");
+//        int choice = scn.nextInt();
+int choice;
+
+
+       do {
+           choice = mainMenu();
+            if(choice == 1){
+                checkIn(capsules, capsulesArray);
+            }else if(choice == 2){
+                  checkOut(capsules, capsulesArray);
+            }
+            else if(choice == 3){
+                viewRooms(capsulesArray);
+            }else if(choice == 4){
+                isFinished = exit();
+            }
+     }while (!isFinished);
+
+
+    }
+
+    //  private static Random rng = new Random();
+
+    public static int capsuleCount() {
+        Scanner scan1 = new Scanner(System.in);
+        System.out.println("How many capsules are there?");
+        String count = scan1.nextLine();
+
+       int userInt = Integer.parseInt(count);
+
+        return userInt;
+    }
+
+
+    public static int mainMenu() {
+
+        System.out.println("Guest Menu");
+        System.out.println("====================");
+        System.out.println("1. Check in");
+        System.out.println("2. Check out");
+        System.out.println("3. View Guests");
+        System.out.println("4. Exit");
+        System.out.println("Choose an option [1-4]: ");
+
+        Scanner scan1 = new Scanner(System.in);
+
+        return scan1.nextInt();
+
+
+    }
+
+//    public static int getIntInRange(){
+//
+//        int choice = Integer.MIN_VALUE;
+//        Scanner scan1 = new Scanner(System.in);
+//
+//        choice = scan1.nextInt();
+//        do {
+//            switch (choice) {
+//                case 1:
+//                    checkIn();
+//                    break;
+//                case 2:
+//                    checkOut();
+//                    break;
+//                case 3:
+//                    viewRooms();
+//                    break;
+//                case 4:
+//                    exit();
+//                    break;
+//                default:
+//                    System.out.println("Unknown action");
+//                    choice = scan1.nextInt();
+//            }
+//            while (!(choice == 4)) ;
+//
+//        }
+//        return choice;
+//
+//        }
+
+    public static String name() {
+        Scanner scan2 = new Scanner(System.in);
+        System.out.println("Guest's name: ");
+        String guestName = scan2.nextLine();
+
+        return guestName;
+    }
+
+
+    public static void checkIn(int numCapsules, String[] capsulesArray) {
+        int chosenRoom = 0;
+        Scanner scan3 = new Scanner(System.in);
```

Comment:
NOT ESSENTIAL: Generating separate scanners solves the issue of the `.nextInt()` >> `.nextLine()` bug, and this is a clever way around for sure, but it seems a bit overwrought. We could get away with generating only one new scanner, allowing it to remain dedicated to String responses, and then using `Integer.parseInt(scan.nextLine())` to get a number value